{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['/usr/share/fonts/urw-base35/NimbusSans-BoldItalic.otf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/Ubuntu-LI.ttf', '/usr/share/fonts/overpass/Overpass-Regular.ttf', '/usr/share/fonts/gnu-free/FreeSansBoldOblique.ttf', '/usr/share/fonts/urw-base35/NimbusMonoPS-Italic.otf', '/usr/share/fonts/sil-padauk/Padauk.ttf', '/usr/share/fonts/dejavu/DejaVuSerifCondensed-BoldItalic.ttf', '/usr/share/fonts/nhn-nanum/NanumGothicBold.ttf', '/usr/share/fonts/lohit-nepali/Lohit-Nepali.ttf', '/usr/share/fonts/google-crosextra-caladea/Caladea-Regular.ttf', '/usr/share/fonts/open-sans/OpenSans-LightItalic.ttf', '/usr/share/fonts/google-crosextra-carlito/Carlito-Italic.ttf', '/usr/share/fonts/dejavu/DejaVuSans-Bold.ttf', '/usr/share/fonts/urw-base35/NimbusSansNarrow-Regular.otf', '/usr/share/fonts/urw-base35/D050000L.otf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-ExtraLightIt.ttf', '/usr/share/fonts/nhn-nanum/NanumGothic.ttf', '/usr/share/fonts/urw-base35/URWGothic-Demi.otf', '/usr/share/fonts/sil-abyssinica/AbyssinicaSIL-R.ttf', '/usr/share/fonts/dejavu/DejaVuSerifCondensed-Bold.ttf', '/usr/share/fonts/urw-base35/NimbusSans-Bold.otf', '/usr/share/fonts/urw-base35/C059-Roman.otf', '/usr/share/fonts/urw-base35/NimbusMonoPS-Regular.otf', '/usr/share/fonts/dejavu/DejaVuSerif-BoldItalic.ttf', '/usr/share/fonts/urw-base35/URWBookman-DemiItalic.otf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-Regular.ttf', '/usr/share/fonts/lohit-tamil/Lohit-Tamil.ttf', '/usr/share/fonts/dejavu/DejaVuSans.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/Ubuntu-C.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/Ubuntu-BI.ttf', '/usr/share/fonts/google-crosextra-carlito/Carlito-Regular.ttf', '/usr/share/fonts/overpass/Overpass-ExtraLight Italic.ttf', '/usr/share/fonts/gnu-free/FreeSerif.ttf', '/usr/share/fonts/urw-base35/NimbusRoman-BoldItalic.otf', '/usr/share/fonts/khmeros/KhmerOS_sys.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/Ubuntu-B.ttf', '/usr/share/fonts/open-sans/OpenSans-BoldItalic.ttf', '/usr/share/fonts/urw-base35/NimbusMonoPS-Bold.otf', '/usr/share/fonts/dejavu/DejaVuSerif.ttf', '/usr/share/fonts/liberation/LiberationSerif-Bold.ttf', '/usr/share/fonts/dejavu/DejaVuSansCondensed-BoldOblique.ttf', '/usr/share/fonts/lohit-assamese/Lohit-Assamese.ttf', '/usr/share/fonts/gnu-free/FreeSerifItalic.ttf', '/usr/share/fonts/gnu-free/FreeMonoOblique.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-BlackIt.ttf', '/usr/share/fonts/dejavu/DejaVuSansMono-Bold.ttf', '/usr/share/fonts/open-sans/OpenSans-ExtraBold.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-ExtraLight.ttf', '/usr/share/fonts/thai-scalable/Waree.ttf', '/usr/share/fonts/sil-padauk/Padauk-bold.ttf', '/usr/share/fonts/urw-base35/URWGothic-DemiOblique.otf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/Ubuntu-MI.ttf', '/usr/share/fonts/lohit-kannada/Lohit-Kannada.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-It.ttf', '/usr/share/fonts/smc/Meera.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/UbuntuMono-BI.ttf', '/usr/share/fonts/urw-base35/URWGothic-BookOblique.otf', '/usr/share/fonts/gnu-free/FreeSerifBold.ttf', '/usr/share/fonts/lohit-devanagari/Lohit-Devanagari.ttf', '/usr/share/fonts/urw-base35/NimbusRoman-Regular.otf', '/usr/share/fonts/dejavu/DejaVuSansCondensed-Bold.ttf', '/usr/share/fonts/liberation/LiberationSans-BoldItalic.ttf', '/usr/share/fonts/lohit-gujarati/Lohit-Gujarati.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-Medium.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/Ubuntu-M.ttf', '/usr/share/fonts/google-crosextra-carlito/Carlito-BoldItalic.ttf', '/usr/share/fonts/dejavu/DejaVuSans-BoldOblique.ttf', '/usr/share/fonts/overpass/Overpass-Light.ttf', '/usr/share/fonts/open-sans/OpenSans-SemiboldItalic.ttf', '/usr/share/fonts/gnu-free/FreeMono.ttf', '/usr/share/fonts/nhn-nanum/NanumGothicExtraBold.ttf', '/usr/share/fonts/stix/STIX-Bold.otf', '/usr/share/fonts/paratype-pt-sans/PTS75F.ttf', '/usr/share/fonts/urw-base35/NimbusMonoPS-BoldItalic.otf', '/usr/share/fonts/open-sans/OpenSans-ExtraBoldItalic.ttf', '/usr/share/fonts/google-crosextra-caladea/Caladea-Bold.ttf', '/usr/share/fonts/open-sans/OpenSans-Bold.ttf', '/usr/share/fonts/lohit-malayalam/Lohit-Malayalam.ttf', '/usr/share/fonts/dejavu/DejaVuSerif-Italic.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-BoldIt.ttf', '/usr/share/fonts/stix/STIX-BoldItalic.otf', '/usr/share/fonts/wqy-microhei/wqy-microhei.ttc', '/usr/share/fonts/paratype-pt-sans/PTN77F.ttf', '/usr/share/fonts/vlgothic/VL-Gothic-Regular.ttf', '/usr/share/fonts/urw-base35/P052-BoldItalic.otf', '/usr/share/fonts/gnu-free/FreeMonoBold.ttf', '/usr/share/fonts/urw-base35/NimbusRoman-Italic.otf', '/usr/share/fonts/open-sans/OpenSans-Light.ttf', '/usr/share/fonts/gnu-free/FreeSansOblique.ttf', '/usr/share/fonts/overpass/Overpass-Light-Italic.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-Bold.ttf', '/usr/share/fonts/paratype-pt-sans/PTS56F.ttf', '/usr/share/fonts/liberation/LiberationMono-Italic.ttf', '/usr/share/fonts/dejavu/DejaVuSerif-Bold.ttf', '/usr/share/fonts/liberation/LiberationMono-Bold.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/Ubuntu-RI.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/UbuntuMono-RI.ttf', '/usr/share/fonts/urw-base35/NimbusSansNarrow-Bold.otf', '/usr/share/fonts/lohit-telugu/Lohit-Telugu.ttf', '/usr/share/fonts/jomolhari/Jomolhari-alpha3c-0605331.ttf', '/usr/share/fonts/urw-base35/Z003-MediumItalic.otf', '/usr/share/fonts/urw-base35/NimbusRoman-Bold.otf', '/usr/share/fonts/thai-scalable/Waree-Bold.ttf', '/usr/share/fonts/paratype-pt-sans/PTS55F.ttf', '/usr/share/fonts/urw-base35/C059-Bold.otf', '/usr/share/fonts/dejavu/DejaVuSans-Oblique.ttf', '/usr/share/fonts/cjkuni-uming/uming.ttc', '/usr/share/fonts/sil-nuosu/NuosuSIL.ttf', '/usr/share/fonts/wqy-zenhei/wqy-zenhei.ttc', '/usr/share/fonts/google-crosextra-caladea/Caladea-BoldItalic.ttf', '/usr/share/fonts/abattis-cantarell/Cantarell-Bold.otf', '/usr/share/fonts/dejavu/DejaVuSansCondensed.ttf', '/usr/share/fonts/urw-base35/NimbusSans-Italic.otf', '/usr/share/fonts/liberation/LiberationSans-Regular.ttf', '/usr/share/fonts/urw-base35/C059-Italic.otf', '/usr/share/fonts/liberation/LiberationMono-BoldItalic.ttf', '/usr/share/fonts/lohit-marathi/Lohit-Marathi.ttf', '/usr/share/fonts/abattis-cantarell/Cantarell-BoldOblique.otf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/Ubuntu-R.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-Light.ttf', '/usr/share/fonts/google-noto-emoji/NotoColorEmoji.ttf', '/usr/share/fonts/urw-base35/URWBookman-Light.otf', '/usr/share/fonts/thai-scalable/Waree-BoldOblique.ttf', '/usr/share/fonts/open-sans/OpenSans-Semibold.ttf', '/usr/share/fonts/gnu-free/FreeSansBold.ttf', '/usr/share/fonts/urw-base35/C059-BdIta.otf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/Inconsolata-Bold.ttf', '/usr/share/fonts/urw-base35/P052-Roman.otf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-SemiboldIt.ttf', '/usr/share/fonts/madan/madan.ttf', '/usr/share/fonts/liberation/LiberationSans-Italic.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-LightIt.ttf', '/usr/share/fonts/google-crosextra-carlito/Carlito-Bold.ttf', '/usr/share/fonts/urw-base35/NimbusSansNarrow-Oblique.otf', '/usr/share/fonts/dejavu/DejaVuSerifCondensed-Italic.ttf', '/usr/share/fonts/stix/STIX-Italic.otf', '/usr/share/fonts/dejavu/DejaVuSerifCondensed.ttf', '/usr/share/fonts/liberation/LiberationSans-Bold.ttf', '/usr/share/fonts/urw-base35/P052-Bold.otf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-MediumIt.ttf', '/usr/share/fonts/abattis-cantarell/Cantarell-Oblique.otf', '/usr/share/fonts/dejavu/DejaVuSansMono.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/DejaVuSans.ttf', '/usr/share/fonts/urw-base35/NimbusSansNarrow-BoldOblique.otf', '/usr/share/fonts/paktype-naskh-basic/PakTypeNaskhBasic.ttf', '/usr/share/fonts/dejavu/DejaVuSansCondensed-Oblique.ttf', '/usr/share/fonts/dejavu/DejaVuSans-ExtraLight.ttf', '/usr/share/fonts/google-crosextra-caladea/Caladea-Italic.ttf', '/usr/share/fonts/lohit-punjabi/Lohit-Punjabi.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/Inconsolata-Regular.ttf', '/usr/share/fonts/liberation/LiberationSerif-Italic.ttf', '/usr/share/fonts/urw-base35/URWGothic-Book.otf', '/usr/share/fonts/overpass/Overpass-Bold.ttf', '/usr/share/fonts/open-sans/OpenSans-Regular.ttf', '/usr/share/fonts/lklug/lklug.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/Ubuntu-Th.ttf', '/usr/share/fonts/gnu-free/FreeSans.ttf', '/usr/share/fonts/dejavu/DejaVuSansMono-BoldOblique.ttf', '/usr/share/fonts/lohit-oriya/Lohit-Oriya.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/UbuntuMono-B.ttf', '/usr/share/fonts/paratype-pt-sans/PTN57F.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-Black.ttf', '/usr/share/fonts/urw-base35/P052-Italic.otf', '/usr/share/fonts/open-sans/OpenSans-Italic.ttf', '/usr/share/fonts/liberation/LiberationMono-Regular.ttf', '/usr/share/fonts/abattis-cantarell/Cantarell-Regular.otf', '/usr/share/fonts/google-noto-emoji/NotoEmoji-Regular.ttf', '/usr/share/fonts/gnu-free/FreeSerifBoldItalic.ttf', '/usr/share/fonts/urw-base35/URWBookman-Demi.otf', '/usr/share/fonts/liberation/LiberationSerif-Regular.ttf', '/usr/share/fonts/stix/STIX-Regular.otf', '/usr/share/fonts/urw-base35/URWBookman-LightItalic.otf', '/usr/share/fonts/lohit-bengali/Lohit-Bengali.ttf', '/usr/share/fonts/khmeros/KhmerOS_content.ttf', '/usr/share/fonts/gnu-free/FreeMonoBoldOblique.ttf', '/usr/share/fonts/overpass/Overpass-Bold-Italic.ttf', '/usr/share/fonts/paratype-pt-sans/PTS76F.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/Ubuntu-L.ttf', '/usr/share/fonts/liberation/LiberationSerif-BoldItalic.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/SourceCodePro-Semibold.ttf', '/usr/share/fonts/overpass/Overpass-Regular-Italic.ttf', '/usr/share/fonts/overpass/Overpass-ExtraLight.ttf', '/usr/share/fonts/thai-scalable/Waree-Oblique.ttf', '/usr/share/fonts/dejavu/DejaVuSansMono-Oblique.ttf', '/usr/share/fonts/urw-base35/NimbusSans-Regular.otf', '/usr/share/fonts/khmeros/KhmerOS.ttf', '/home/goldpm1/miniconda3/envs/cnvpytor/fonts/UbuntuMono-R.ttf']\n"
     ]
    }
   ],
   "source": [
    "import matplotlib.font_manager\n",
    "\n",
    "# Get a list of available font names\n",
    "font_list = matplotlib.font_manager.findSystemFonts(fontpaths=None, fontext='ttf')\n",
    "\n",
    "print (font_list)\n",
    "\n",
    "# # Print the font names\n",
    "# for font_path in font_list:\n",
    "#     font_info = matplotlib.font_manager.FontProperties(fname=font_path)\n",
    "#     print(font_info.get_name())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "************** Reported Current Memory Use: 0.38 GB *****************\n",
      "\n",
      "Extracting signature 2 for mutation type 96\n",
      "The matrix normalizing cutoff is 9600\n",
      "\n",
      "\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-34674106cc75>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     21\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     22\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 23\u001b[0;31m sig.sigProfilerExtractor( \"matrix\", \n\u001b[0m\u001b[1;32m     24\u001b[0m                                     \u001b[0;34m\"/data/project/Alzheimer/CLEMENT/03.combinedoutput/3.BioData/Moore_1D/adrenal_gland_zona_glomerulosa/PD28690-L1/SigProfiler/output/DeNovo\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     25\u001b[0m                                      \u001b[0;34m\"/data/project/Alzheimer/CLEMENT/03.combinedoutput/3.BioData/Moore_1D/adrenal_gland_zona_glomerulosa/PD28690-L1/SigProfiler/output/SBS/BioData.SBS96.all\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/Yonsei/python/3.8.1/lib/python3.8/site-packages/SigProfilerExtractor/sigpro.py\u001b[0m in \u001b[0;36msigProfilerExtractor\u001b[0;34m(input_type, output, input_data, reference_genome, opportunity_genome, cosmic_version, context_type, exome, minimum_signatures, maximum_signatures, nmf_replicates, resample, batch_size, cpu, gpu, nmf_init, precision, matrix_normalization, seeds, min_nmf_iterations, max_nmf_iterations, nmf_test_conv, nmf_tolerance, nnls_add_penalty, nnls_remove_penalty, initial_remove_penalty, collapse_to_SBS96, clustering_distance, export_probabilities, make_decomposition_plots, stability, min_stability, combined_stability, allow_stability_drop, get_all_signature_matrices)\u001b[0m\n\u001b[1;32m    748\u001b[0m             \u001b[0mconverge_information\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;31m \u001b[0m\u001b[0;31m\\\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    749\u001b[0m             \u001b[0mreconstruction_error\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;31m \u001b[0m\u001b[0;31m\\\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 750\u001b[0;31m             processes = sub.decipher_signatures(execution_parameters,\n\u001b[0m\u001b[1;32m    751\u001b[0m                                                 \u001b[0mgenomes\u001b[0m\u001b[0;34m=\u001b[0m \u001b[0mgenomes\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    752\u001b[0m                                                 \u001b[0mmut_context\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mm\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/Yonsei/python/3.8.1/lib/python3.8/site-packages/SigProfilerExtractor/subroutines.py\u001b[0m in \u001b[0;36mdecipher_signatures\u001b[0;34m(execution_parameters, genomes, i, totalIterations, cpu, mut_context, noise_rep_pair)\u001b[0m\n\u001b[1;32m    541\u001b[0m             \u001b[0mresults\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mW\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mH\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0msimilarities\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    542\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 543\u001b[0;31m         results = parallel_runs(execution_parameters, genomes=genomes, totalProcesses=totalProcesses, \\\n\u001b[0m\u001b[1;32m    544\u001b[0m             verbose = False, replicate_generators=replicate_generators)\n\u001b[1;32m    545\u001b[0m     \u001b[0mtoc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/Yonsei/python/3.8.1/lib/python3.8/site-packages/SigProfilerExtractor/subroutines.py\u001b[0m in \u001b[0;36mparallel_runs\u001b[0;34m(execution_parameters, genomes, totalProcesses, verbose, replicate_generators)\u001b[0m\n\u001b[1;32m    487\u001b[0m                 \u001b[0mresample\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mresample\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minit\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0minit\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnormalization_cutoff\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mnormalization_cutoff\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;31m \u001b[0m\u001b[0;31m\\\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    488\u001b[0m                 norm=norm, gpu=gpu, execution_parameters=execution_parameters)\n\u001b[0;32m--> 489\u001b[0;31m         \u001b[0mresult_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpool\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpool_nmf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbatch_generator_pair\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    490\u001b[0m         \u001b[0mpool\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    491\u001b[0m         \u001b[0mpool\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/Yonsei/python/3.8.1/lib/python3.8/multiprocessing/pool.py\u001b[0m in \u001b[0;36mmap\u001b[0;34m(self, func, iterable, chunksize)\u001b[0m\n\u001b[1;32m    362\u001b[0m         \u001b[0;32min\u001b[0m \u001b[0ma\u001b[0m \u001b[0mlist\u001b[0m \u001b[0mthat\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mreturned\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    363\u001b[0m         '''\n\u001b[0;32m--> 364\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_map_async\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfunc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0miterable\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmapstar\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mchunksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    365\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    366\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mstarmap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0miterable\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mchunksize\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/Yonsei/python/3.8.1/lib/python3.8/multiprocessing/pool.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    760\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    761\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 762\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    763\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mready\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    764\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mTimeoutError\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/Yonsei/python/3.8.1/lib/python3.8/multiprocessing/pool.py\u001b[0m in \u001b[0;36mwait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    757\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    758\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mwait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 759\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_event\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    760\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    761\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/Yonsei/python/3.8.1/lib/python3.8/threading.py\u001b[0m in \u001b[0;36mwait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    556\u001b[0m             \u001b[0msignaled\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_flag\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    557\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0msignaled\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 558\u001b[0;31m                 \u001b[0msignaled\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_cond\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    559\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0msignaled\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    560\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/Yonsei/python/3.8.1/lib/python3.8/threading.py\u001b[0m in \u001b[0;36mwait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    300\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m    \u001b[0;31m# restore state no matter what (e.g., KeyboardInterrupt)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    301\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mtimeout\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 302\u001b[0;31m                 \u001b[0mwaiter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0macquire\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    303\u001b[0m                 \u001b[0mgotit\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    304\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# from SigProfilerMatrixGenerator.scripts import SigProfilerMatrixGeneratorFunc as matGen\n",
    "\n",
    "# #help ( matGen )\n",
    "\n",
    "# # matrices = matGen.SigProfilerMatrixGeneratorFunc ( project = \"BioData\", reference_genome = \"GRCh37\", path_to_input_files = \"/data/project/Alzheimer/CLEMENT/03.combinedoutput/3.BioData/Moore_1D/adrenal_gland_zona_glomerulosa/PD28690-L1/SigProfilerMatrixGenerator\", \n",
    "# #                                                                                     exome=False, bed_file=None, chrom_based=False, plot=True, tsb_stat=False, seqInfo=True)\n",
    "\n",
    "# from SigProfilerAssignment import Analyzer as Analyze\n",
    "# #help ( Analyze )\n",
    "# Analyze.cosmic_fit(\"/data/project/Alzheimer/CLEMENT/03.combinedoutput/3.BioData/Moore_1D/adrenal_gland_zona_glomerulosa/PD28690-L1/SigProfilerMatrixGenerator/output/SBS/BioData.SBS96.all\", \n",
    "#                                 \"/data/project/Alzheimer/CLEMENT/03.combinedoutput/3.BioData/Moore_1D/adrenal_gland_zona_glomerulosa/PD28690-L1/SigProfilerMatrixGenerator/output/Assignment\", \n",
    "#                                 input_type=\"matrix\", \n",
    "#                                 cosmic_version=3.3, exome=False,\n",
    "#                                 genome_build=\"GRCh37\", signature_database=None,\n",
    "#                                 exclude_signature_subgroups=None, export_probabilities=False,\n",
    "#                                 export_probabilities_per_mutation=False, make_plots=False,\n",
    "#                                 sample_reconstruction_plots=False, verbose=False)\n",
    "\n",
    "\n",
    "from SigProfilerExtractor import sigpro as sig\n",
    "\n",
    "\n",
    "sig.sigProfilerExtractor( \"matrix\", \n",
    "                                    \"/data/project/Alzheimer/CLEMENT/03.combinedoutput/3.BioData/Moore_1D/adrenal_gland_zona_glomerulosa/PD28690-L1/SigProfiler/output/DeNovo\",\n",
    "                                     \"/data/project/Alzheimer/CLEMENT/03.combinedoutput/3.BioData/Moore_1D/adrenal_gland_zona_glomerulosa/PD28690-L1/SigProfiler/output/SBS/BioData.SBS96.all\", \n",
    "                                     reference_genome=\"GRCh37\", opportunity_genome = \"GRCh37\", context_type = \"default\", exome = False, \n",
    "                                    minimum_signatures=2, maximum_signatures=7, nmf_replicates=50, resample = True, batch_size=1, cpu= 20, gpu=False, \n",
    "                                    nmf_init=\"random\", precision= \"single\", matrix_normalization= \"gmm\", seeds= \"random\", \n",
    "                                    min_nmf_iterations= 10000, max_nmf_iterations=1000000, nmf_test_conv= 10000, nmf_tolerance= 1e-15, get_all_signature_matrices= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(518444, 14)\n",
      "clone 0 : (6, 11)\n",
      "clone 1 : (8, 11)\n",
      "clone 2 : (21, 11)\n",
      "clone 3 : (243, 11)\n",
      "clone 4 : (534, 11)\n",
      "clone 5 : (499, 11)\n",
      "clone whole : (1311, 11)\n"
     ]
    }
   ],
   "source": [
    "def GenerateMatrix (Moore_VCF, Clone_no, **kwargs):\n",
    "    import pandas as pd\n",
    "    \n",
    "    Moore_VCF = Moore_VCF.reset_index(drop = True)\n",
    "    SigProfMatrix = pd.DataFrame (index = range ( len (Moore_VCF)),  columns = [\"Project\", \"Sample\", \"ID\", \"Genome\", \"mut_type\", \"chrom\",\t\"pos_start\",\"pos_end\",\t\"ref\",\t\"alt\"\t,\"Type\"]) \n",
    "    SigProfMatrix[\"Project\"] = \"BioData\"\n",
    "    SigProfMatrix[\"Sample\"] = str(Clone_no)\n",
    "    SigProfMatrix[\"ID\"] = Moore_VCF[\"pos\"]\n",
    "    SigProfMatrix[\"Genome\"] = \"GRCh37\"\n",
    "    SigProfMatrix[\"mut_type\"] = \"SNP\"\n",
    "    SigProfMatrix[\"chrom\"] = Moore_VCF[\"chr\"]\n",
    "    SigProfMatrix[\"pos_start\"] = Moore_VCF[\"start\"]\n",
    "    SigProfMatrix[\"pos_end\"] = Moore_VCF[\"end\"]\n",
    "    SigProfMatrix[\"ref\"] = Moore_VCF[\"ref\"]\n",
    "    SigProfMatrix[\"alt\"] = Moore_VCF[\"alt\"]\n",
    "    SigProfMatrix[\"Type\"] = \"SOMATIC\"\n",
    "    \n",
    "    SigProfMatrix = SigProfMatrix.drop_duplicates()\n",
    "\n",
    "    SigProfMatrix.to_csv( kwargs[\"OUTPUT_PATH\"], index=False, header = True,  sep=\"\\t\")\n",
    "    print (\"clone {} : {}\".format (Clone_no, SigProfMatrix.shape) )\n",
    "\n",
    "\n",
    "def extract (Moore_VCF, **kwargs):\n",
    "    import numpy as np\n",
    "\n",
    "    pos_whole = []\n",
    "\n",
    "    # Clone number 대로  mutation_id를 뽑아주기\n",
    "    with open( kwargs [\"DECISION_MEMBERSHIP_PATH\"], \"r\") as input_file:\n",
    "        membership = np.array( [int(line.strip()) for line in input_file] )\n",
    "        npvaf = pd.read_csv ( kwargs [\"NPVAF_PATH\"], sep = \"\\t\")\n",
    "        for Clone_no in sorted(np.unique ( membership) ) :\n",
    "            npvaf_Clone_no =  (npvaf.iloc [np.where(membership == Clone_no)[0] ])\n",
    "            pos_Clone_no =  list ( npvaf_Clone_no.iloc[:, 0] )\n",
    "            pos_whole += pos_Clone_no\n",
    "\n",
    "            Moore_VCF_Clone_no = Moore_VCF[Moore_VCF['pos'].isin( pos_Clone_no ) == True]\n",
    "            \n",
    "            kwargs[\"OUTPUT_PATH\"] = kwargs[\"OUTPUT_DIR\"] + \"/clone{}.txt\".format(Clone_no)\n",
    "\n",
    "            #print (Moore_VCF_Clone_no.head())\n",
    "            GenerateMatrix (Moore_VCF_Clone_no, Clone_no, **kwargs)\n",
    "\n",
    "    # whole\n",
    "    kwargs[\"OUTPUT_PATH\"] = kwargs[\"OUTPUT_DIR\"] + \"/whole.txt\"\n",
    "    Moore_VCF_whole = Moore_VCF[Moore_VCF['pos'].isin( pos_whole ) == True]\n",
    "    GenerateMatrix (Moore_VCF_whole, \"whole\", **kwargs)\n",
    "\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    from SigProfilerMatrixGenerator.scripts import SigProfilerMatrixGeneratorFunc as matGen\n",
    "    import pandas as pd\n",
    "    import argparse\n",
    "\n",
    "    Moore_VCF = pd.read_csv(\"/data/project/Alzheimer/CLEMENT/resource/paper/whole_info.txt\", sep = \"\\t\")\n",
    "\n",
    "    # column을 소문자로 바꾸기\n",
    "    Moore_VCF.rename(columns=lambda x: x.lower(), inplace=True)\n",
    "\n",
    "    # lexicographically 정렬하기\n",
    "    custom_order = [\"1\", \"2\", \"3\", \"4\", \"5\", \"6\", \"7\", \"8\", \"9\", \"10\", \"11\", \"12\", \"13\", \"14\", \"15\", \"16\", \"17\", \"18\", \"19\", \"20\", \"21\", \"22\", \"X\", \"Y\"]\n",
    "    Moore_VCF['chr_categorical'] = pd.Categorical(Moore_VCF['chr'], categories=custom_order, ordered=True)\n",
    "    Moore_VCF = Moore_VCF.sort_values(by= ['chr_categorical', 'start']).reset_index(drop=True).drop (\"chr_categorical\", axis = 1)\n",
    "\n",
    "    # 앞에 \"chr\" 붙여주기\n",
    "    Moore_VCF[\"pos\"]  = \"chr\" + Moore_VCF[\"chr\"] + \"_\" + Moore_VCF[\"start\"].astype(\"str\")\n",
    "    #print (Moore_VCF.shape)\n",
    "\n",
    "\n",
    "    kwargs = {}\n",
    "    # kwargs [\"DECISION_MEMBERSHIP_PATH\"] =  \"/data/project/Alzheimer/CLEMENT/03.combinedoutput/3.BioData/Moore_1D_AG/glomerulosa_L1/result/CLEMENT_decision.membership.txt\"\n",
    "    # kwargs [\"NPVAF_PATH\"] = \"/data/project/Alzheimer/CLEMENT/02.npvaf/3.BioData/Moore_1D_AG/glomerulosa_L1/npvaf.txt\"\n",
    "    # kwargs[\"OUTPUT_DIR\"] = \"/data/project/Alzheimer/CLEMENT/03.combinedoutput/3.BioData/Moore_1D_AG/glomerulosa_L1/SigProfiler\"\n",
    "\n",
    "    kwargs [\"DECISION_MEMBERSHIP_PATH\"] =  \"/data/project/Alzheimer/YSscript/cle/data/CLEMENT/3.BioData/Moore_2D_AG/fasciculata_L1_glomerulosa_L1/CLEMENT_hard_1st.membership.txt\"\n",
    "    kwargs [\"NPVAF_PATH\"] = \"/data/project/Alzheimer/CLEMENT/02.npvaf/3.BioData/Moore_2D_AG/fasciculata_L1_glomerulosa_L1/npvaf.txt\"\n",
    "    kwargs[\"OUTPUT_DIR\"] = \"/data/project/Alzheimer/CLEMENT/03.combinedoutput/3.BioData/Moore_2D_AG/fasciculata_L1_glomerulosa_L1/SigProfiler\"\n",
    "\n",
    "\n",
    "    extract (Moore_VCF, **kwargs)\n",
    "\n",
    "\n",
    "    #python3 /data/project/Alzheimer/YSscript/cle/3.BioData_pipe1-2.MatrixGenerator.py --OUTPUT_DIR \"/data/project/Alzheimer/CLEMENT/03.combinedoutput/3.BioData/Moore_2D_AG/fasciculata_L1_glomerulosa_L1/SigProfiler\"\n",
    "    #python3 /data/project/Alzheimer/YSscript/cle/3.BioData_pipe1-3.Assignment.py --OUTPUT_SBS96 /data/project/Alzheimer/CLEMENT/03.combinedoutput/3.BioData/Moore_2D_AG/fasciculata_L1_glomerulosa_L1/SigProfiler/output/SBS/BioData.SBS96.all --ASSIGNMENT_DIR /data/project/Alzheimer/CLEMENT/03.combinedoutput/3.BioData/Moore_2D_AG/fasciculata_L1_glomerulosa_L1/SigProfiler/output/Assignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from SigProfilerMatrixGenerator.scripts import SigProfilerMatrixGeneratorFunc as matGen\n",
    "import pandas as pd\n",
    "\n",
    "Moore_VCF = pd.read_csv(\"/data/project/Alzheimer/CLEMENT/resource/paper/whole_info.txt\", sep = \"\\t\")\n",
    "\n",
    "# column을 소문자로 바꾸기\n",
    "Moore_VCF.rename(columns=lambda x: x.lower(), inplace=True)\n",
    "\n",
    "# lexicographically 정렬하기\n",
    "custom_order = [\"1\", \"2\", \"3\", \"4\", \"5\", \"6\", \"7\", \"8\", \"9\", \"10\", \"11\", \"12\", \"13\", \"14\", \"15\", \"16\", \"17\", \"18\", \"19\", \"20\", \"21\", \"22\", \"X\", \"Y\"]\n",
    "Moore_VCF['chr_categorical'] = pd.Categorical(Moore_VCF['chr'], categories=custom_order, ordered=True)\n",
    "Moore_VCF = Moore_VCF.sort_values(by= ['chr_categorical', 'start']).reset_index(drop=True).drop (\"chr_categorical\", axis = 1)\n",
    "\n",
    "# 앞에 \"chr\" 붙여주기\n",
    "Moore_VCF[\"pos\"]  = \"chr\" + Moore_VCF[\"chr\"] + \"_\" + Moore_VCF[\"start\"].astype(\"str\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>depth_count</th>\n",
       "      <th>alt_count</th>\n",
       "      <th>vaf</th>\n",
       "      <th>sample</th>\n",
       "      <th>donorid</th>\n",
       "      <th>tissue</th>\n",
       "      <th>func</th>\n",
       "      <th>gene</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>115672</td>\n",
       "      <td>115672</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>13</td>\n",
       "      <td>8</td>\n",
       "      <td>0.615385</td>\n",
       "      <td>PD43850u_P50_RTM_B10</td>\n",
       "      <td>PD43850</td>\n",
       "      <td>colon_crypt</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>OR4F5,LOC729737</td>\n",
       "      <td>chr1_115672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>525383</td>\n",
       "      <td>525383</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>PD28690bw_APP_3_D5</td>\n",
       "      <td>PD28690</td>\n",
       "      <td>appendix_crypt</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>OR4F3,OR4F29</td>\n",
       "      <td>chr1_525383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>540719</td>\n",
       "      <td>540719</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>15</td>\n",
       "      <td>11</td>\n",
       "      <td>0.733333</td>\n",
       "      <td>PD43851t_P52_PRST_F11</td>\n",
       "      <td>PD42565</td>\n",
       "      <td>prostate_acinus</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>OR4F3,OR4F29</td>\n",
       "      <td>chr1_540719</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chr   start     end ref alt  depth_count  alt_count       vaf  \\\n",
       "0   1  115672  115672   C   T           13          8  0.615385   \n",
       "1   1  525383  525383   G   A            5          5  1.000000   \n",
       "2   1  540719  540719   C   G           15         11  0.733333   \n",
       "\n",
       "                  sample  donorid           tissue        func  \\\n",
       "0   PD43850u_P50_RTM_B10  PD43850      colon_crypt  intergenic   \n",
       "1     PD28690bw_APP_3_D5  PD28690   appendix_crypt  intergenic   \n",
       "2  PD43851t_P52_PRST_F11  PD42565  prostate_acinus  intergenic   \n",
       "\n",
       "              gene          pos  \n",
       "0  OR4F5,LOC729737  chr1_115672  \n",
       "1     OR4F3,OR4F29  chr1_525383  \n",
       "2     OR4F3,OR4F29  chr1_540719  "
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter rows where 'pos' is in the list 'a'\n",
    "list_of_interest = [\"chr1_115672\", \"chr1_525383\", \"chr1_540719\"]\n",
    "Moore_VCF = Moore_VCF[Moore_VCF['pos'].isin( list_of_interest ) == True]\n",
    "Moore_VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Project</th>\n",
       "      <th>Sample</th>\n",
       "      <th>ID</th>\n",
       "      <th>Genome</th>\n",
       "      <th>mut_type</th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos_start</th>\n",
       "      <th>pos_end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BioData</td>\n",
       "      <td>clone1</td>\n",
       "      <td>chr1_115672</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1</td>\n",
       "      <td>115672</td>\n",
       "      <td>115672</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SOMATIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BioData</td>\n",
       "      <td>clone1</td>\n",
       "      <td>chr1_525383</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1</td>\n",
       "      <td>525383</td>\n",
       "      <td>525383</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SOMATIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BioData</td>\n",
       "      <td>clone1</td>\n",
       "      <td>chr1_540719</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1</td>\n",
       "      <td>540719</td>\n",
       "      <td>540719</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SOMATIC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Project  Sample           ID  Genome mut_type chrom  pos_start  pos_end  \\\n",
       "0  BioData  clone1  chr1_115672  GRCh37      SNP     1     115672   115672   \n",
       "1  BioData  clone1  chr1_525383  GRCh37      SNP     1     525383   525383   \n",
       "2  BioData  clone1  chr1_540719  GRCh37      SNP     1     540719   540719   \n",
       "\n",
       "  ref alt     Type  \n",
       "0   C   T  SOMATIC  \n",
       "1   G   A  SOMATIC  \n",
       "2   C   G  SOMATIC  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SigProfMatrix = pd.DataFrame (index = range ( len (Moore_VCF)),  columns = [\"Project\", \"Sample\", \"ID\", \"Genome\", \"mut_type\", \"chrom\",\t\"pos_start\",\"pos_end\",\t\"ref\",\t\"alt\"\t,\"Type\"]) \n",
    "SigProfMatrix[\"Project\"] = \"BioData\"\n",
    "SigProfMatrix[\"Sample\"] = \"clone1\"\n",
    "SigProfMatrix[\"ID\"] = Moore_VCF[\"pos\"]\n",
    "SigProfMatrix[\"Genome\"] = \"GRCh37\"\n",
    "SigProfMatrix[\"mut_type\"] = \"SNP\"\n",
    "SigProfMatrix[\"chrom\"] = Moore_VCF[\"chr\"]\n",
    "SigProfMatrix[\"pos_start\"] = Moore_VCF[\"start\"]\n",
    "SigProfMatrix[\"pos_end\"] = Moore_VCF[\"end\"]\n",
    "SigProfMatrix[\"ref\"] = Moore_VCF[\"ref\"]\n",
    "SigProfMatrix[\"alt\"] = Moore_VCF[\"alt\"]\n",
    "SigProfMatrix[\"Type\"] = \"SOMATIC\"\n",
    "\n",
    "SigProfMatrix"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
