{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from SigProfilerMatrixGenerator.scripts import SigProfilerMatrixGeneratorFunc as matGen\n",
    "import pandas as pd\n",
    "\n",
    "Moore_VCF = pd.read_csv(\"/data/project/Alzheimer/CLEMENT/resource/paper/whole_info.txt\", sep = \"\\t\")\n",
    "\n",
    "# column을 소문자로 바꾸기\n",
    "Moore_VCF.rename(columns=lambda x: x.lower(), inplace=True)\n",
    "\n",
    "# lexicographically 정렬하기\n",
    "custom_order = [\"1\", \"2\", \"3\", \"4\", \"5\", \"6\", \"7\", \"8\", \"9\", \"10\", \"11\", \"12\", \"13\", \"14\", \"15\", \"16\", \"17\", \"18\", \"19\", \"20\", \"21\", \"22\", \"X\", \"Y\"]\n",
    "Moore_VCF['chr_categorical'] = pd.Categorical(Moore_VCF['chr'], categories=custom_order, ordered=True)\n",
    "Moore_VCF = Moore_VCF.sort_values(by= ['chr_categorical', 'start']).reset_index(drop=True).drop (\"chr_categorical\", axis = 1)\n",
    "\n",
    "# 앞에 \"chr\" 붙여주기\n",
    "Moore_VCF[\"pos\"]  = \"chr\" + Moore_VCF[\"chr\"] + \"_\" + Moore_VCF[\"start\"].astype(\"str\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>depth_count</th>\n",
       "      <th>alt_count</th>\n",
       "      <th>vaf</th>\n",
       "      <th>sample</th>\n",
       "      <th>donorid</th>\n",
       "      <th>tissue</th>\n",
       "      <th>func</th>\n",
       "      <th>gene</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>115672</td>\n",
       "      <td>115672</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>13</td>\n",
       "      <td>8</td>\n",
       "      <td>0.615385</td>\n",
       "      <td>PD43850u_P50_RTM_B10</td>\n",
       "      <td>PD43850</td>\n",
       "      <td>colon_crypt</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>OR4F5,LOC729737</td>\n",
       "      <td>chr1_115672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>525383</td>\n",
       "      <td>525383</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>PD28690bw_APP_3_D5</td>\n",
       "      <td>PD28690</td>\n",
       "      <td>appendix_crypt</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>OR4F3,OR4F29</td>\n",
       "      <td>chr1_525383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>540719</td>\n",
       "      <td>540719</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>15</td>\n",
       "      <td>11</td>\n",
       "      <td>0.733333</td>\n",
       "      <td>PD43851t_P52_PRST_F11</td>\n",
       "      <td>PD42565</td>\n",
       "      <td>prostate_acinus</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>OR4F3,OR4F29</td>\n",
       "      <td>chr1_540719</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chr   start     end ref alt  depth_count  alt_count       vaf  \\\n",
       "0   1  115672  115672   C   T           13          8  0.615385   \n",
       "1   1  525383  525383   G   A            5          5  1.000000   \n",
       "2   1  540719  540719   C   G           15         11  0.733333   \n",
       "\n",
       "                  sample  donorid           tissue        func  \\\n",
       "0   PD43850u_P50_RTM_B10  PD43850      colon_crypt  intergenic   \n",
       "1     PD28690bw_APP_3_D5  PD28690   appendix_crypt  intergenic   \n",
       "2  PD43851t_P52_PRST_F11  PD42565  prostate_acinus  intergenic   \n",
       "\n",
       "              gene          pos  \n",
       "0  OR4F5,LOC729737  chr1_115672  \n",
       "1     OR4F3,OR4F29  chr1_525383  \n",
       "2     OR4F3,OR4F29  chr1_540719  "
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter rows where 'pos' is in the list 'a'\n",
    "list_of_interest = [\"chr1_115672\", \"chr1_525383\", \"chr1_540719\"]\n",
    "Moore_VCF = Moore_VCF[Moore_VCF['pos'].isin( list_of_interest ) == True]\n",
    "Moore_VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Project</th>\n",
       "      <th>Sample</th>\n",
       "      <th>ID</th>\n",
       "      <th>Genome</th>\n",
       "      <th>mut_type</th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos_start</th>\n",
       "      <th>pos_end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BioData</td>\n",
       "      <td>clone1</td>\n",
       "      <td>chr1_115672</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1</td>\n",
       "      <td>115672</td>\n",
       "      <td>115672</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SOMATIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BioData</td>\n",
       "      <td>clone1</td>\n",
       "      <td>chr1_525383</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1</td>\n",
       "      <td>525383</td>\n",
       "      <td>525383</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SOMATIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BioData</td>\n",
       "      <td>clone1</td>\n",
       "      <td>chr1_540719</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1</td>\n",
       "      <td>540719</td>\n",
       "      <td>540719</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SOMATIC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Project  Sample           ID  Genome mut_type chrom  pos_start  pos_end  \\\n",
       "0  BioData  clone1  chr1_115672  GRCh37      SNP     1     115672   115672   \n",
       "1  BioData  clone1  chr1_525383  GRCh37      SNP     1     525383   525383   \n",
       "2  BioData  clone1  chr1_540719  GRCh37      SNP     1     540719   540719   \n",
       "\n",
       "  ref alt     Type  \n",
       "0   C   T  SOMATIC  \n",
       "1   G   A  SOMATIC  \n",
       "2   C   G  SOMATIC  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SigProfMatrix = pd.DataFrame (index = range ( len (Moore_VCF)),  columns = [\"Project\", \"Sample\", \"ID\", \"Genome\", \"mut_type\", \"chrom\",\t\"pos_start\",\"pos_end\",\t\"ref\",\t\"alt\"\t,\"Type\"]) \n",
    "SigProfMatrix[\"Project\"] = \"BioData\"\n",
    "SigProfMatrix[\"Sample\"] = \"clone1\"\n",
    "SigProfMatrix[\"ID\"] = Moore_VCF[\"pos\"]\n",
    "SigProfMatrix[\"Genome\"] = \"GRCh37\"\n",
    "SigProfMatrix[\"mut_type\"] = \"SNP\"\n",
    "SigProfMatrix[\"chrom\"] = Moore_VCF[\"chr\"]\n",
    "SigProfMatrix[\"pos_start\"] = Moore_VCF[\"start\"]\n",
    "SigProfMatrix[\"pos_end\"] = Moore_VCF[\"end\"]\n",
    "SigProfMatrix[\"ref\"] = Moore_VCF[\"ref\"]\n",
    "SigProfMatrix[\"alt\"] = Moore_VCF[\"alt\"]\n",
    "SigProfMatrix[\"Type\"] = \"SOMATIC\"\n",
    "\n",
    "SigProfMatrix"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
